src/bsoncxx/json.cpp

```diff
@@ -274,7 +274,12 @@ std::string to_json(types::value value) {
 
     json_visitor v(ss, false, 0);
 
-    switch ((int)value.type()) {
+    if (!value.type()) {
+        // TODO: use bsoncxx error category
+        throw std::runtime_error("invalid value");
+    }
+
+    switch (static_cast<typename std::underlying_type<bsoncxx::type>::type>(*value.type())) {
```

Comment:
It's not actually an enum, it's a macro, defined here: https://github.com/mongodb/mongo-cxx-driver/blob/master/src/bsoncxx/enums/type.hpp. So if the *value.type() is not cast to an int, it tries to cast all the ints in that file to value_type, which fails.